substrate/frame/democracy/src/lib.rs

```diff
@@ -234,9 +240,14 @@ pub mod pallet {
 		/// The Preimage provider.
 		type Preimages: QueryPreimage<H = Self::Hashing> + StorePreimage;
 
-		/// Currency type for this pallet.
-		type Currency: ReservableCurrency<Self::AccountId>
-			+ LockableCurrency<Self::AccountId, Moment = BlockNumberFor<Self>>;
+		/// The overarching hold reason.
+		type RuntimeHoldReason: From<HoldReason>;
+
+		/// The fungible trait.
+		type Fungible: fungible::Inspect<Self::AccountId>
+			+ fungible::Mutate<Self::AccountId>
+			+ fungible::MutateHold<Self::AccountId, Reason = Self::RuntimeHoldReason>
+			+ fungible::hold::Balanced<Self::AccountId>;
```

Comment:
might need to use freeze instead of hold here, 
will investigate and update this